invenio_drafts_resources/services/records/components/media_files.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020-2021 CERN.
+# Copyright (C) 2021 Northwestern University.
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Drafts-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Records service component base classes."""
+from invenio_i18n import gettext as _
+from invenio_records_resources.services.records.components import (
+    MediaFilesOptionsComponent,
+)
+from marshmallow import ValidationError
+
+from .files import DraftFilesComponent
+
+
+class DraftMediaFilesComponent(DraftFilesComponent):
+    """Draft media files component."""
+
+    _files_attr_key = "media_files"
+    _files_data_key = "media_files"
+    _files_bucket_attr_key = "media_bucket"
+    _files_bucket_id_attr_key = "media_bucket_id"
+
+    def __init__(self, service, *args, **kwargs):
+        """Constructor."""
+        super().__init__(service)
+        self.files_component = MediaFilesOptionsComponent(service)
+
+    def create(self, identity, data=None, record=None, errors=None):
+        """Assigns files.enabled.
+
+        NOTE: `record` actually refers to the draft
+              (this interface is used in records-resources and rdm-records)
+        """
+        draft = record
+        files = self.get_record_files(draft)
+
+        # disable the media files by default
+        enabled = data.get(self.files_data_key, {}).get("enabled", False)
+
+        files.enabled = enabled
+
+    def update_draft(self, identity, data=None, record=None, errors=None):
```

Comment:
This method and the files component `update_draft` are formed 90% the same code, IIUC the only difference is the `enabled` (here is defaults to the record's and in the other component to `True`. Then in the other component, there is a permission check (based on `enabled`), the subsequent code lines 60-92 are the same than lines 95-119. This can be extracted the private methods (one with the enabled and checks over it, which can be different in each component and one for the rest of the logic) and remove duplicated code.